_templates/controller.go.tmpl

```diff
@@ -10,33 +11,25 @@ import (
 
 	"github.com/gin-gonic/gin"
 	"github.com/jinzhu/gorm"
+	"github.com/serenize/snaker"
 )
 
-func set{{ .Model.Name }}Preload(fields []string, db *gorm.DB) ([]string, *gorm.DB) {
-	sel := make([]string, len(fields))
-	copy(sel, fields)
-	offset := 0
-	for key, val := range fields {
-		switch val {
-{{ range .Model.Fields }}{{ if .IsAssociation }}
-		case "{{ .JSONName }}":
-			db = db.Preload("{{ .Name }}")
-			sel = append(sel[:(key-offset)], sel[(key+1-offset):]...)
-			offset += 1
-			idflag := true
-			for _, v := range sel {
-				if v == "{{ if .IsBelongsTo }}{{ .JSONName }}_id{{ else }}id{{ end }}" {
-					idflag = false
-					break
-				}
-			}
-			if idflag {
-				sel = append(sel, "{{ if .IsBelongsTo }}{{ .JSONName }}_id{{ else }}id{{ end }}")
-			}
-{{ end }}{{ end }}
+func set{{ .Model.Name }}Preloads(preloads string, db *gorm.DB) *gorm.DB {
```

Comment:
Currently, this is just same function between models.
So why not rename this to `setPreloads` and put this to appropriate place?